### PR TITLE
Enable Non-API Deploys

### PR DESCRIPTION
**Pull request by brolewis**
A lot of little changes. Tried to make a bunch of little changes to make it more palatable CbC.

### CODE REVIEW

File: .github/workflows/staging-deploy.yml [hunk 1]
```diff
@@ -2,25 +2,30 @@ name: staging-deploy
 on:
   workflow_call:
     inputs:
-      environment_url:
+      project_name:
```
Comment (brolewis) [inline .github/workflows/staging-deploy.yml:5]: I know that moving `api_name` to `project_name` is a breaking change for the 3 existing APIs, but makes sense as half the potential use cases aren't APIs

File: python-setup/action.yml [hunk 2]
```diff
@@ -21,6 +21,9 @@ runs:
       with:
         fetch-depth: ${{ inputs.fetch_depth }}
 
+    - name: Set up QEMU
+      uses: docker/setup-qemu-action@v1
```
Comment (brolewis) [inline python-setup/action.yml:25]: This step only takes a few seconds to execute, so thought it made sense to be safe and always include it

**Review by bradsbrown — APPROVED** [review]
Looks good, will merge pending the open question

File: .github/workflows/staging-deploy.yml (hunk 1 above)
Comment (brolewis) [reply .github/workflows/staging-deploy.yml:5]: Correct, but I've got pending commits for ALL THE ~THINGS~ REPOS to quickly follow this merging.

File: .github/workflows/pull-requests.yml [hunk 3]
```diff
@@ -63,6 +75,11 @@ jobs:
               - 'run'
               - '.github/workflows/pull-requests.yml'
 
+      - uses: andstor/file-existence-action@v1
+        id: check_files
+        with:
+          files: "tests/conftest.py"
```
Comment (brolewis) [inline .github/workflows/pull-requests.yml:81]: As I was adding this functionality for when to run the tests, I realized that we could use this action to determine if deployed tests should be run (`"tests/deploy_tests/*"`), and when we should build docs (`"docs/*"`), and even when to update IAM permissions (`"*/support/aws_iam.py"`). I didn't want to codify that until I got feedback that it was a good direction to go.

Comment (brolewis) [reply .github/workflows/pull-requests.yml:71]: Ended up making the change for deploy tests and building docs, but IAM permissions felt like a bridge too far for me.

**Review by bradsbrown — APPROVED** [review]

**Pull request merged**